### PR TITLE
ui: Allow font-awesome icon usage and optimise icon size inconsistency

### PR DESCRIPTION
**Pull request by rohityadavcloud**
This addresses that issue that plugins customisation in the UI (https://docs.cloudstack.apache.org/en/4.19.1.1/adminguide/ui.html#basic-ui-c
ustomization) isn't able to use all the ant-design-vue icons.

We already use and support fontawesome components such as for Kubernetes menu etc.
This extends the usage of fontawesome (free) solid icons for custom plugins
section in config.json. The plugin can be defined as array:

```
  "plugins": [
    {
        "name": "HomeLab",
        "icon": ["fa-solid", "fa-server"],
        "path": "https://dash.yadav.cloud/dash/"
    }
  ],
```

Here the icons is declared as a json array config, where the first
parameter is the pack/type name ("fa-solid") and the second parameter is
the icons name with the "fa-" prefix.

The solid icon can be searched and used from here:
https://fontawesome.com/search?q=server&o=r&s=solid

This also fixes icon size inconsistency when fa os-logo icons are used versus when resource icons are used. This also changes some of the infocard icons to use better icons from font awesome.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI
- [ ] test (unit or integration test code)

### Testing

With the example json snipped defined above, would result in this which uses a font awesome solid icon:
Screenshot 2024-09-28 at 12 38 03 PM

### CODE REVIEW

**Comment by rohityadavcloud** [comment]
FYI - this is from Swen (proIO's) email from users ML who found a limitation in plugins/UI icons usage.

@blueorangutan ui

**Comment by rohityadavcloud** [comment]
QA'd manually this seems to be working okay: (notice the icons)
Screenshot 2024-09-30 at 1 13 43 PM

**Comment by DaanHoogland** [comment]
> FYI - this is from Swen (proIO's) email from users ML who found a limitation in plugins/UI icons usage.

https://lists.apache.org/thread/0062c2o0dl4vps562bdkpmw0ym8f5yq6

**Comment by DaanHoogland** [comment]
@JoaoJandre , will you consider this?

**Comment by JoaoJandre** [comment]
@rohityadavcloud @DaanHoogland If I understood correctly, this PR is improving which fonts can be used when creating new plugins, and not fixing any bugs. If so, does it need to go into 4.20? It seems to me that if it gets merged as soon as 4.20 is out, new plugins can already start using the fonts, so there would be no need to merge it now. 

Please correct me if I'm mistaken.

**Comment by DaanHoogland** [comment]
> @rohityadavcloud @DaanHoogland If I understood correctly, this PR is improving which fonts can be used when creating new plugins, and not fixing any bugs. If so, does it need to go into 4.20? It seems to me that if it gets merged as soon as 4.20 is out, new plugins can already start using the fonts, so there would be no need to merge it now.
> 
> Please correct me if I'm mistaken.

I think you are right, I understood it as "We cannot add our plugin because of a bug" but let's move this forward then.

File: ui/src/components/header/ProjectMenu.vue [hunk 1]
```diff
@@ -81,7 +81,7 @@ export default {
       const projects = []
       const getNextPage = () => {
         this.loading = true
-        api('listProjects', { listAll: true, page: page, pageSize: 500, showIcon: true }).then(json => {
+        api('listProjects', { listAll: true, page: page, pageSize: 500, details: 'min', showIcon: true }).then(json => {
```
Comment (GutoVeronezi) [inline ui/src/components/header/ProjectMenu.vue:84]: @rohityadavcloud, how this line is related to the issue and the PR's proposal?

Comment (rohityadavcloud) [reply ui/src/components/header/ProjectMenu.vue:84]: Something I spotted, makes the listing faster. We use details=min in many other places.

**Comment by DaanHoogland** [comment]
@rohityadavcloud , i resolved a conflict here. Hope it is alright like this by you.

**Comment by DaanHoogland** [comment]
@blueorangutan ui

**Review by vishesh92 — APPROVED** [review]
clgtm. Verified changes in UI for List View and info card.

**Review by shwstppr — APPROVED** [review]
LGTM. Checked different views, looking fine to me

**Comment by rohityadavcloud** [comment]
Thanks @DaanHoogland @vishesh92 @shwstppr 
cc @sbrueseke

**Comment by sbrueseke** [comment]
LGTM, thx everybody for the effort! :-)

**Pull request merged**